### ISSUE TITLE
Channel force-closed and closing failed

### ISSUE THREAD

**Issue by bonomat** [get10101/10101#1056]
In preparation for upgrading to 1.2.0 I've tested the following: 

1. start coordinator, maker, app in 1.1.0
2. open a position
3. upgrade coordinator, maker, app to https://github.com/get10101/10101/pull/1018

After 3. the channel in the app was force-closed. 

```
2023-08-03 08:43:05 DEBUG lightning::ln::peer_handler: Got Err message from 03dd76f4346a6e685a9d1d68b44c83321a89c97a45c66e718a5e54115c37e8b7ba: Invalid commitment tx signature from peer    
```

The broadcast of the transaction failed though: 

```
2023-08-03 08:47:48  INFO ln_dlc_node::ldk_node_wallet: Broadcasting transaction txid=e70298a0bd2a1108b7073ab0ba0038ef72c3378abb3260104e104739f48d06b3 raw_tx=020000000001013aa062355362a0bab20074fcc0632e87cbd38a96c4fc530f4678fbb361b4b5ff000000000005a3068002e0b400000000000022002002c1756fd84bd0e87049cb69a515125c3479d293df3947f516a34e497bb0be774db8000000000000160014a69171880048584ef117eb1edd5afdfc9f2647a9040047304402206cbc8d479acba4b7ab807f52ca312755537adf6e641dee6feb69d6e46396484402202e76d591ef71c5721e4c6a88f84daf219302902728c82d4e1c913d59c05d78e40147304402201cf5f79942d083a23f289b82b045ff47252eeb8d0ed0124e673ae466cb1690a1022027896da77675fc7a5c8bdb3c1f6606adf9ba38ef3423856a33dfdb33822b5877014752210228d3632c2d6ad8eae53d370f192544c88732cf67719536cb2499cc1f1caa341a21023b3729e076d7f59bd2f8b3c62985f581f69d7768cabbcdb5d591b26668b0fef252ae8a4e2e20
```

**Comment by bonomat** [comment]
Full logs: 

```
2023-08-03 08:43:05 DEBUG lightning::ln::channelmanager: Received channel_update for channel c3d6415498c11323d7eb8c31902b9de33e07c02e2cf050a52c7f6b1299849278.    
2023-08-03 08:43:05 DEBUG lightning::ln::peer_handler: Error handling message from 03dd76f4346a6e685a9d1d68b44c83321a89c97a45c66e718a5e54115c37e8b7ba; ignoring: Couldn't find channel for update    
2023-08-03 08:43:05 DEBUG lightning::ln::peer_handler: Handling UpdateHTLCs event in peer_handler for node 03dd76f4346a6e685a9d1d68b44c83321a89c97a45c66e718a5e54115c37e8b7ba with 0 adds, 0 fulfills, 0 fails for channel c3d6415498c11323d7eb8c31902b9de33e07c02e2cf050a52c7f6b1299849278    
2023-08-03 08:43:05 TRACE lightning_background_processor: Done persisting ChannelManager.    
2023-08-03 08:43:05 TRACE lightning_background_processor: Persisting ChannelManager...    
2023-08-03 08:43:05 TRACE lightning_background_processor: Done persisting ChannelManager.    
2023-08-03 08:43:05 DEBUG lightning::ln::peer_handler: Got Err message from 03dd76f4346a6e685a9d1d68b44c83321a89c97a45c66e718a5e54115c37e8b7ba: Invalid commitment tx signature from peer    
2023-08-03 08:43:05 ERROR lightning::ln::channelmanager: Force-closing channel c3d6415498c11323d7eb8c31902b9de33e07c02e2cf050a52c7f6b1299849278    
2023-08-03 08:43:05 DEBUG lightning::ln::channelmanager: Finishing force-closure of channel with 0 HTLCs to fail    
2023-08-03 08:43:05 TRACE lightning::chain::chainmonitor: Updating ChannelMonitor for channel c3d6415498c11323d7eb8c31902b9de33e07c02e2cf050a52c7f6b1299849278    
2023-08-03 08:43:05  INFO lightning::chain::channelmonitor: Applying update to monitor c3d6415498c11323d7eb8c31902b9de33e07c02e2cf050a52c7f6b1299849278, bringing update_id from 19 to 20 with 1 changes.    
2023-08-03 08:43:05 TRACE lightning::chain::channelmonitor: Updating ChannelMonitor: channel force closed, should broadcast: true    
2023-08-03 08:43:05 DEBUG lightning::chain::channelmonitor: Getting signed latest holder commitment transaction!    
2023-08-03 08:43:05  INFO lightning::chain::channelmonitor: Broadcasting local commitment tx with txid e70298a0bd2a1108b7073ab0ba0038ef72c3378abb3260104e104739f48d06b3    
2023-08-03 08:43:05  INFO ln_dlc_node::ldk_node_wallet: Broadcasting transaction txid=e70298a0bd2a1108b7073ab0ba0038ef72c3378abb3260104e104739f48d06b3 raw_tx=020000000001013aa062355362a0bab20074fcc0632e87cbd38a96c4fc530f4678fbb361b4b5ff000000000005a3068002e0b400000000000022002002c1756fd84bd0e87049cb69a515125c3479d293df3947f516a34e497bb0be774db8000000000000160014a69171880048584ef117eb1edd5afdfc9f2647a9040047304402206cbc8d479acba4b7ab807f52ca312755537adf6e641dee6feb69d6e46396484402202e76d591ef71c5721e4c6a88f84daf219302902728c82d4e1c913d59c05d78e40147304402201cf5f79942d083a23f289b82b045ff47252eeb8d0ed0124e673ae466cb1690a1022027896da77675fc7a5c8bdb3c1f6606adf9ba38ef3423856a33dfdb33822b5877014752210228d3632c2d6ad8eae53d370f192544c88732cf67719536cb2499cc1f1caa341a21023b3729e076d7f59bd2f8b3c62985f581f69d7768cabbcdb5d591b26668b0fef252ae8a4e2e20
2023-08-03 08:43:05 ERROR ln_dlc_node::ldk_node_wallet: Failed to broadcast transaction: Esplora client error: HttpResponse(400)
2023-08-03 08:43:05 DEBUG lightning::chain::chainmonitor: Persistence of ChannelMonitorUpdate for channel c3d6415498c11323d7eb8c31902b9de33e07c02e2cf050a52c7f6b1299849278 completed    
2023-08-03 08:43:05 TRACE lightning_background_processor: Persisting ChannelManager...    
2023-08-03 08:43:05 TRACE lightning_background_processor: Done persisting ChannelManager.    
2023-08-03 08:43:05  INFO ln_dlc_node::ln::event_handler: Received event event=ChannelClosed { channel_id: [195, 214, 65, 84, 152, 193, 19, 35, 215, 235, 140, 49, 144, 43, 157, 227, 62, 7, 192, 46, 44, 240, 80, 165, 44, 127, 107, 18, 153, 132, 146, 120], user_channel_id: 158740726855705437111138918903816125822, reason: CounterpartyForceClosed { peer_msg: "Invalid commitment tx signature from peer" } }
2023-08-03 08:43:05  INFO ln_dlc_node::ln::event_handler: Channel closed user_channel_id=776c5a17-b9af-4634-b4b3-15991f07fd7e chan
```

**Comment by holzeis** [comment]
Maybe related to https://github.com/get10101/10101/issues/874

**Comment by luckysori** [comment]
I ran into a different problem which appears to be fixed by #1061 anyway.

In my case, after the upgrade there is no force-closure but the two parties disagree on the status of the open position, so they can't make progress.
